File: src/com/beyondbell/connectFour/Board.java
package com.beyondbell.connectFour;

import org.jetbrains.annotations.Contract;

final class Board {
	private final Piece[][] board = {
			{Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty},
			{Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty},
			{Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty},
			{Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty},
			{Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty},
			{Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty, Piece.Empty}
	};

	@Contract(pure = true)
	final byte getRowCount() {
		return (byte) board.length;
	}

	@Contract(pure = true)
	final byte getColumnCount() {
		return (byte) board[0].length;
	}

	@Contract(pure = true)
	final Piece getPiece(final byte row, final byte column) {
		return board[row][column];
	}

	final void setPiece(final byte row, final byte column, final Piece piece) {
		board[row][column] = piece;
	}

	/**
	 * Checks if the board is full. Should be run after checking for a specific win since it returns either full or empty.
	 *
	 * @return Whether or Not the Board is Full
	 */
	@Contract(pure = true)
	final boolean isFull() {
		for (byte column = 0; column < board[0].length; column++) { // If Top is Full, Entire Board Must be Full
			if (board[0][column] == Piece.Empty) {
				return false;
			}
		}
		return true;
	}
}

File: src/main/java/com/beyondbell/connectFourAPI/Board.kt
package com.beyondbell.connectFourAPI

import com.beyondbell.connectFourAPI.Piece.Empty

class Board {
	private val board = arrayOf(
			arrayOf(Empty, Empty, Empty, Empty, Empty, Empty, Empty),
			arrayOf(Empty, Empty, Empty, Empty, Empty, Empty, Empty),
			arrayOf(Empty, Empty, Empty, Empty, Empty, Empty, Empty),
			arrayOf(Empty, Empty, Empty, Empty, Empty, Empty, Empty),
			arrayOf(Empty, Empty, Empty, Empty, Empty, Empty, Empty),
			arrayOf(Empty, Empty, Empty, Empty, Empty, Empty, Empty)
	)

	fun getRowCount() : Int {
		return board.size
	}

	fun getColumnCount() : Int {
		return board[0].size
	}

	fun isFull() : Boolean {
		for (column in 0 until board[0].size) {
			if (board[0][column] == Empty) {
				return false
			}
		}
		return true
	}

	fun getPiece(row: Int, column: Int): Piece {
		return board[row][column]
	}

	fun setPiece(row: Int, column: Int, piece: Piece) {
		board[row][column] = piece
	}
}